x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/capacity/AutoscalingCapacity.java

```diff
@@ -29,18 +32,28 @@ public class AutoscalingCapacity implements ToXContent, Writeable {
     public static class AutoscalingResources implements ToXContent, Writeable {
         private final ByteSizeValue storage;
         private final ByteSizeValue memory;
+        private final Integer processors;
```

Comment:
I agree that current capacity can for now only return int. But we could request a float vCPUs and with the way cloud calculates quota this seems to make sense. Ideally we should then follow up with letting cloud pass the true amount of vCPUs (as a float) that we are getting into ES. This might be useful for proper sizing of thread pools too.

This is likely to be important for ingest based autoscaling (and I suppose ML scaling too).

Reply:
I think it makes sense to return a `float` here. We expect a float (or a range of floats) in the desired nodes https://github.com/elastic/elasticsearch/blob/eb8c4ba97b41a0ad41c535d642d462616e91decc/server/src/main/java/org/elasticsearch/cluster/metadata/DesiredNode.java#L103-L104

Additionally, other systems such as [k8s](https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/#cpu-units) accept fractional values to specify the number of requested CPUs.

Reply:
Okey dokey. Switched it to a `float`.